File: beecrowd/2654.cpp
#include <iostream>
using namespace std;

int main()
{
    int candidatos;
    cin >> candidatos;

    string deus;
    int maiorPoder, maiorDeuses, maiorMortes;

    cin >> deus >> maiorPoder >> maiorDeuses >> maiorMortes;

    for (int i = 0; i <= candidatos; i++)
    {
        string nome;
        int poder, deuses, mortes;

        cin >> nome >> poder >> deuses >> mortes;

        if ((poder > maiorPoder) || (poder == maiorPoder && deuses > maiorDeuses) || (poder == maiorPoder && deuses == maiorDeuses && mortes < maiorMortes))
        {
            deus = nome;
            maiorPoder = poder;
            maiorDeuses = deuses;
            maiorMortes = mortes;
        }
        else if (poder == maiorPoder && deuses == maiorDeuses && mortes == maiorMortes)
        {
            bool novo = false;
            if (nome.length() <= deus.length())
            {
                for (int j = 0; j < nome.length(); j++)
                {
                    if (nome[j] > deus[j])
                    {
                        break;
                    }
                    else if (nome[j] < deus[j])
                    {
                        novo = true;
                    }
                }
            }
            else
            {
                for (int j = 0; j < deus.length(); j++)
                {
                    if (nome[j] > deus[j])
                    {
                        break;
                    }
                    else if (nome[j] < deus[j])
                    {
                        novo = true;
                    }
                }
            }
            if (novo)
            {
                deus = nome;
                maiorPoder = poder;
                maiorDeuses = deuses;
                maiorMortes = mortes;
            }
        }
    }
    cout << deus << "\n";

    return 0;
}

File: beecrowd/2654.py
# -*- coding: utf-8 -*-

deuses = int(input())

nome, poder, matou, morreu = input().split()

maioresAtributos = {
    "nome": nome,
    "poder": int(poder),
    "matou": int(matou),
    "morreu": int(morreu)
}

for i in range(deuses - 1):
    nome, poder, matou, morreu = input().split()
    poder, matou, morreu = [int(j) for j in [poder, matou, morreu]]

    maiorPoder = poder > maioresAtributos["poder"]
    matouMais = poder == maioresAtributos["poder"] and matou > maioresAtributos["matou"]
    morreuMenos = poder == maioresAtributos["poder"] and matou == maioresAtributos["matou"] and morreu < maioresAtributos["morreu"]
    iguaisNomeMenor = poder == maioresAtributos["poder"] and matou == maioresAtributos["matou"] and morreu == maioresAtributos["morreu"] and nome < maioresAtributos["nome"]
        

    if maiorPoder or matouMais or morreuMenos or iguaisNomeMenor:
        maioresAtributos = {
            "nome": nome,
            "poder": int(poder),
            "matou": int(matou),
            "morreu": int(morreu)
        }
    
print(maioresAtributos["nome"])
